lib/browser/extensions/checksum.js

```diff
@@ -0,0 +1,39 @@
+class Checksum {
+  static supportedAlogrithms = ['SHA-1', 'SHA-256', 'SHA-384', 'SHA-512'];
+
+  constructor (algo = 'SHA-256') {
+    this.algo = this._resolveAlgo(algo)
+  }
+
+  /**
+   * Resolves algorithm to one of the inbuilt types
+   * @param {String} algo contains user provided checksumAlgo option
+   */
+  _resolveAlgo = (algo) => {
+    const resolvedAlgo = Checksum.supportedAlgorithms.find(supportedAlgo => supportedAlgo === algo.toUpperCase())
+    if (!resolvedAlgo) {
+      throw new Error(
+        `tus: unsupported checksumAlgo provided. Supported values are : ${Checksum.supportedAlgorithms.join(',')}`,
+      )
+    }
+    return resolvedAlgo
+  }
+  /**
+   * Gets Hexadecimal digest using the algorithm set in this.algo
+   * @param {ArrayBuffer} data contains the chunk of data to be hashed
+   */
+
+  getHexDigest = async (data) => {
+    try {
+      const hashBuffer = await crypto.subtle.digest(this.algo, data)
```

Comment:
A method for checking if the Crypto API is support would be great, so the users gets notified early if that's not the case.

Reply:
Done, added.